src/proxy-animation.js

```diff
@@ -818,7 +814,7 @@ export class ProxyAnimation {
       effect: null,
       // Range when using a view-timeline. The default range is cover 0% to
       // 100%.
-      timeRange: null,
+      timeRange: timeline instanceof ScrollTimeline ? timeline.timeRange : null,
```

Comment:
timeRange cannot be a member of timeline since multiple animations can link to the same timeline with different ranges. Also we avoid hanging properties directly off of timeline that are not explicitly part of the API for encapsulation.  That's why we use the animation and timeline maps.  the map links the animation or timeline object to an object for storing additional information that is not exposed to the end user.

Reply:
Updated, I pass timeRange to ProxyAnimation constructor and save it in proxyAnimations map.